src/engine/idTable/CompressedExternalIdTable.h

```diff
@@ -338,7 +344,11 @@ class CompressedExternalIdTableBase {
   }
   // Add a single row to the input. The type of `row` needs to be something that
   // can be `push_back`ed to a `IdTable`.
-  void push(const auto& row) requires requires { currentBlock_.push_back(row); }
+  CPP_template(typename R)(
+      requires HasPushBackConcept<IdTableStatic<NumStaticCols>,
```

Comment:
```suggestion
      requires HasPushBackConcept<decltype(currentBlock_),
```
I find this a little more readable here, but I have no strong opinion.